forc-pkg/src/pkg.rs

```diff
@@ -824,7 +771,23 @@ fn dep_path(
             )
         }
         SourcePinned::Registry(_reg) => unreachable!("registry dependencies not yet supported"),
-        SourcePinned::Root => unreachable!("a `Root` node cannot be a dependency"),
+        SourcePinned::Member => {
+            // If a node has a root dependency it is a member of the workspace.
+            let parent_workspace_manifest = node_manifest
+                .dir()
+                .parent()
+                .and_then(|path| WorkspaceManifestFile::from_dir(path).ok());
+            if let Some(parent_workspace_manifest) = parent_workspace_manifest {
+                parent_workspace_manifest
+                    .members()
+                    .zip(parent_workspace_manifest.member_paths()?)
+                    .find(|(member_name, _)| *member_name == dep_name)
+                    .map(|(_, member_path)| member_path)
+                    .ok_or_else(|| anyhow!("cannot find dependency in the workspace"))
+            } else {
+                bail!("a node cannot depend on a root node if it is not a member of a workspace")
+            }
+        }
```

Comment:
Nice work! That said, it's a shame we have to load the workspace manifest file via I/O again here. I'll leave some thoughts on manifest file reading in the top comment.